pacman/utilities/json_utils.py

```diff
@@ -283,57 +335,125 @@ def vertex_add_contstraints_from_json(json_dict, graph):
     graph = MachineGraph(json_dict.get("label"))
     for j_vertex in json_dict["vertices"]:
-        graph.add_vertex(vertex_from_json(j_vertex, convert_constraints=False))
+        graph.add_vertex(machine_vertex_from_json(
+            j_vertex, convert_constraints=False))
     # Only do constraints when we have all the vertexes to link to
     for j_vertex in json_dict["vertices"]:
         vertex_add_contstraints_from_json(j_vertex, graph)
-    for j_edge in json_dict["edges"]:
-        edge = edge_from_json(j_edge, graph)
-        graph.add_edge(edge, "JSON_MOCK")
+    for j_partitions in json_dict["partitions"]:
+        partition_from_json(j_partitions, graph)
     return graph
 
 
+def graphs_to_json(application_graph, machine_graph, graph_mapper):
+    """
+
+    :param ApplicationGraph application_graph:
+    :param MachineGraph machine_graph:
+    :param graph_mapper:  TODO NUKE ME!
+    :return:
+    """
+    json_dict = OrderedDict()
+    try:
+        if machine_graph.label is not None:
+            json_dict["label"] = machine_graph.label
+        json_list = []
+        for application_vertex in application_graph.vertices:
+            json_list.append(application_vertex_to_json(
+                application_vertex, graph_mapper))
+        json_dict["vertices"] = json_list
+        json_list = []
+        for partition in machine_graph.outgoing_edge_partitions:
+            json_list.append(partition_to_json(partition))
+        json_dict["partitions"] = json_list
+    except Exception as ex:  # pylint: disable=broad-except
+        json_dict["exception"] = str(ex)
+    return json_dict
+
+
+def graphs_from_json(json_dict):
+    json_dict = json_to_object(json_dict)
+    application_graph = ApplicationGraph(json_dict.get("label"))
+    machine_graph = MachineGraph(json_dict.get("label"))
+    graph_mapper = GraphMapper()
+    for j_application_vertex in json_dict["vertices"]:
+        application_vertex = application_vertex_from_json(j_application_vertex)
+        application_graph.add_vertex(application_vertex)
+        for j_machine_vertex in j_application_vertex["machine_vertices"]:
+            machine_vertex = machine_vertex_from_json(
+                j_machine_vertex, convert_constraints=False)
+            machine_graph.add_vertex(machine_vertex)
+            graph_mapper.simple_add_vertex_mapping(
+                machine_vertex, application_vertex)
+    # Only do constraints when we have all the vertexes to link to
+    for j_application_vertex in json_dict["vertices"]:
+        vertex_add_contstraints_from_json(
+            j_application_vertex, application_graph)
+        for j_machine_vertex in j_application_vertex["machine_vertices"]:
+            vertex_add_contstraints_from_json(j_machine_vertex, machine_graph)
+    for j_partitions in json_dict["partitions"]:
+        partition_from_json(j_partitions, machine_graph)
+    return application_graph, machine_graph, graph_mapper
```

Comment:
So far as I can tell, this never adds an edge to the application graph, so it won't be the application graph that went in.  Is this what you wanted?

Reply:
To the best of my knowledge once Machine Edges are created none of the placers, routers, key allocators and their support algorithms ever use Application Edges.

Reply:
I don't think any of the current ones do; note though that if we come up with one that can make use of it, this information will be lost.  If you are happy with "adding this when needed", I am happy to merge this, just so long as we understand what is missing.